.circleci/config.yml

```diff
@@ -285,234 +282,101 @@ commands:
-      name: win/default
-    steps:
-      - win_setup_conda:
-          python_version: "3.7"
-      - win_unit_tests
-
-  win_unit_tests_38:
-    executor:
-      name: win/default
-    steps:
-      - win_setup_conda:
-          python_version: "3.8"
-      - checkout
-      - win_setup_env
-      - restore_cache:
-          key: kedro-deps-v1-win-{{ checksum "requirements.txt" }}-{{ checksum "test_requirements.txt" }}
-      - win_setup_requirements
-      - run:
-          name: Set HDF5_DISABLE_VERSION_CHECK environment variable
-          command: setx /m HDF5_DISABLE_VERSION_CHECK 1
-      - run:
-          name: Run unit tests without Spark and TensorFlow
-          # Run `test_parallel_runner.py` separately because of `Windows fatal exception: stack overflow`
-          command: |
-            conda activate kedro_builder
-            pytest .\tests --ignore .\tests\extras\datasets\spark --ignore .\tests\extras\datasets\tensorflow --ignore tests\framework\session\test_session_hooks.py --ignore .\tests\runner\test_parallel_runner.py --no-cov
-            if ($?) { pytest .\tests\runner\test_parallel_runner.py --no-cov }
-
-  win_e2e_tests_36:
-    executor:
-      name: win/default
-    steps:
-      - win_setup_conda:
-          python_version: "3.6"
-      - win_e2e_tests
-
-  win_e2e_tests_37:
-    executor:
-      name: win/default
-    steps:
-      - win_setup_conda:
-          python_version: "3.7"
-      - win_e2e_tests
-
-  win_e2e_tests_38:
-    executor:
-      name: win/default
-    steps:
-      - win_setup_conda:
-          python_version: "3.8"
-      - win_e2e_tests
-
-  win_pip_compile_36:
-    executor:
-      name: win/default
-    steps:
-      - win_setup_conda:
-          python_version: "3.6"
-      - win_pip_compile
-
-  win_pip_compile_37:
-    executor:
-      name: win/default
-    steps:
-      - win_setup_conda:
-          python_version: "3.7"
-      - win_pip_compile:
-          # Save cache only for Python 3.7. There is no need to save it for each Python.
-          cache_save: true
-
-  win_pip_compile_38:
-    executor:
-      name: win/default
-    steps:
-      - win_setup_conda:
-          python_version: "3.8"
-      - win_pip_compile
-
 workflows:
-  version: 2
+  version: 2.1
   regular:
     jobs:
-      - unit_tests_36
-      - linters_36
-      - e2e_tests_36
-      - docs_36
-      - docs_linkcheck_37
-      - unit_tests_37
-      - linters_37
-      - e2e_tests_37
-      - docs_37
-      - unit_tests_38
-      - linters_38
-      - e2e_tests_38
-      - pip_compile_36
-      - pip_compile_37
-      - pip_compile_38
-      - win_unit_tests_36
-      - win_unit_tests_37
-      - win_unit_tests_38
-      - win_pip_compile_36
-      - win_pip_compile_37
-      - win_pip_compile_38
-      - win_e2e_tests_36
-      - win_e2e_tests_37
-      - win_e2e_tests_38
-      - run_kedro_viz:
+      - e2e_tests:
+          matrix:
+            parameters:
+              python_version: ["3.6", "3.7", "3.8"]
```

Comment:
This effectively does the following:
```
jobs:
  - e2e_tests:
     name: e2e_tests-3.6
     python_version: 3.6
  - e2e_tests:
     name: e2e_tests-3.7
     python_version: 3.7
  - e2e_tests:
     name: e2e_tests-3.8
     python_version: 3.8
```
i.e. invokes the `e2e_tests` job three times, once for each `python_version`.